core/trino-main/src/main/java/io/trino/execution/StageState.java

```diff
@@ -116,6 +116,6 @@ public boolean canScheduleMoreTasks()
                 // completing successfully when it should fail..
                 return true;
         }
-        return true;
+        throw new IllegalStateException();
```

Comment:
Maybe
`
default:
                throw new IllegalStateException();
`
instead ?

Reply:
This way is better as it allows the rule to catch unexhaustive `switch` to trigger.

Reply:
Add some message like `"Unhandled state: " + this`